diff --git a/htdocs/login.R b/htdocs/login.R
index c15cebf..e3cbecc 100644
--- a/htdocs/login.R
+++ b/htdocs/login.R
@@ -35,7 +35,7 @@ run <- function(url, query, body, headers)
     if (length(body) > 2 && "execLogin" %in% body['action']) {
       res <- unlist(rcloud.support:::session.server.auth(realm="rcloud.exec",user=body['user'],pwd=body['pwd']))
       if (length(res) > 2) {
-        extra.headers <- paste0("Set-Cookie: execUser=", res[2], "; domain=", getConf("cookie.domain"),"; path=/;\r\nSet-Cookie: execToken=", res[1], "; domain=", getConf("cookie.domain"), "; path=/;")
+        extra.headers <- c(paste0("Set-Cookie: user=", usr, "; ","path=/;\r\nSet-Cookie: token=", token, "; "," path=/;"), extra.headers)
         cookies$execToken <- res[1]
       } else return(list("<html><head></head><body>Authentication failed - please check your username and password.</body></html>", "text/html"))
     }
@@ -63,8 +63,8 @@ run <- function(url, query, body, headers)
       ## so we can generate a token
       token <- rcloud.support:::generate.token()
       rcloud.support:::set.token(usr, token)
-      extra.headers <- c(paste0("Set-Cookie: user=", usr, "; domain=", getConf("cookie.domain"),"; path=/;\r\nSet-Cookie: token=", token, "; domain=", getConf("cookie.domain"),"; path=/;"), extra.headers)
-      ## re-create the back-end because the username/token have changed
+      extra.headers <- c(paste0("Set-Cookie: user=", usr, "; ","path=/;\r\nSet-Cookie: token=", token, "; "," path=/;"), extra.headers)
+         ## re-create the back-end because the username/token have changed
       ctx <- create.gist.backend(usr, token)
       url <- gist::auth.url(redirect, ctx=ctx)
     }
diff --git a/htdocs/login_successful.R b/htdocs/login_successful.R
index 8879957..90fa99d 100644
--- a/htdocs/login_successful.R
+++ b/htdocs/login_successful.R
@@ -20,6 +20,6 @@ run <- function(url, query, body, headers)
                "<pre>Welcome, ", username, ".</pre>",
                "</body></html>", sep=''),
          "text/html",
-         paste("Set-Cookie: user=", username, "; domain=", rcloud.config("cookie.domain"), "; path=/;\r\nSet-Cookie: token=", token, "; domain=", rcloud.config("cookie.domain"), "; path=/;\r\nRefresh: 0.1; url=", ret, sep=''))
+         paste("Set-Cookie: user=", username, "; ", "path=/;\r\nSet-Cookie: token=", token, "; "," path=/;\r\nRefresh: 0.1; url=", ret, sep=''))
   } else list("<html><head></head><body>Invalid token, could not authenticate with the back-end</body></html>", "text/html")
 }
diff --git a/htdocs/logout.R b/htdocs/logout.R
index 3090f4c..73f8b3a 100644
--- a/htdocs/logout.R
+++ b/htdocs/logout.R
@@ -22,5 +22,5 @@ run <- function(url, query, body, headers) {
   if (is.null(ret)) ret <- '/goodbye.R'
   list("<html><head></head><body>Logout...</body></html>",
        "text/html",
-       paste0("Set-Cookie: user=; domain=", rcloud.config("cookie.domain"), "; path=/; expires=Thu, 01 Jan 1970 00:00:00 GMT\r\nSet-Cookie: token=; domain=", rcloud.config("cookie.domain"), "; path=/; expires=Thu, 01 Jan 1970 00:00:00 GMT\r\nSet-Cookie: execUser=; domain=", rcloud.config("cookie.domain"), "; path=/; expires=Thu, 01 Jan 1970 00:00:00 GMT\r\nSet-Cookie: execToken=; domain=", rcloud.config("cookie.domain"), "; path=/; expires=Thu, 01 Jan 1970 00:00:00 GMT\r\nRefresh: 0.1; url=", ret, sep=''))
+       paste0("Set-Cookie: user=; ", "path=/; expires=Thu, 01 Jan 1970 00:00:00 GMT\r\nSet-Cookie: token=;", " path=/; expires=Thu, 01 Jan 1970 00:00:00 GMT\r\nSet-Cookie: execUser=; "," path=/; expires=Thu, 01 Jan 1970 00:00:00 GMT\r\nSet-Cookie: execToken=; ", " path=/; expires=Thu, 01 Jan 1970 00:00:00 GMT\r\nRefresh: 0.1; url=", ret, sep=''))
 }
